llvm/lib/Target/AMDGPU/AMDGPUCallLowering.cpp

```diff
@@ -63,6 +63,12 @@ struct AMDGPUOutgoingValueHandler : public CallLowering::OutgoingValueHandler {
 
   void assignValueToReg(Register ValVReg, Register PhysReg,
                         const CCValAssign &VA) override {
+    if (VA.getLocVT() == MVT::i1 &&
+        MIRBuilder.getMF().getSubtarget<GCNSubtarget>().isWave64()) {
```

Comment:
The wave32 and wave64 paths shouldn't diverge. The special case is i1, independent of the wavesize

Reply:
Done.